File: src/main/java/org/shark/example/dao/repository/TaskHistoryRepository.java
package org.shark.example.dao.repository;

import org.shark.example.dao.repository.pojo.TaskHistoryDo;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface TaskHistoryRepository extends JpaRepository<TaskHistoryDo, Integer> {

    @Query(value = "select * from TASK_HISTORY order by id desc limit :size offset :offset", nativeQuery = true)
    List<TaskHistoryDo> findByPage(Integer size, Integer offset);
}

File: src/main/kotlin/org/shark/example/dao/repository/TaskHistoryRepository.kt
package org.shark.example.dao.repository

import org.shark.example.dao.repository.pojo.TaskHistoryDo
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface TaskHistoryRepository : JpaRepository<TaskHistoryDo, Int> {

    @Query(value = "select * from TASK_HISTORY order by id desc limit :size offset :offset", nativeQuery = true)
    fun findByPage(size: Int, offset: Int): List<TaskHistoryDo>
}
